File: src/week2/ji/Leet_2375.java
class Solution {
    public String smallestNumber(String pattern) {
        StringBuilder answer = new StringBuilder();
        StringBuilder stack = new StringBuilder();
        for (int i = 0; i <= pattern.length(); i++) {
            stack.append(i + 1);
            if (i == pattern.length() || pattern.charAt(i) == 'I') {
                answer.append(stack.reverse());
                stack = new StringBuilder();
            }
        }

        return answer.toString();
    }
}

File: src/week2/ji/Leet_2375.py
class Solution:
    def smallestNumber(self, pattern: str) -> str:
        answer = ""
        stack = []
        for i in range(len(pattern)+1):
            stack.append(str(i+1))
            if i == len(pattern) or pattern[i] == 'I':
                answer += "".join(stack[::-1])
                stack = []
        return answer
